src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,110 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double GROW_COEFFICIENT = 0.75;
+    private static final byte DEFAULT_SIZE = 10;
+    private int size = DEFAULT_SIZE;
+    private int currentIndex = 0;
+    private T[] storage;
+
+    @SuppressWarnings("unchecked")
+    public ArrayList() {
+        storage = (T[]) new Object[DEFAULT_SIZE];
+    }
+
     @Override
     public void add(T value) {
-
+        shouldGrow();
+        storage[currentIndex++] = value;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > currentIndex || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    String.format("Index: %d, Size: %d", index, currentIndex)
+            );
```

Comment:
You may create a separate method for this check

Reply:
The check uses only once in code, so there's no need to create a function. But i can create a function that throw an error because this piece of code repeats twice in program.